### DISCUSSION TITLE
Json type array_starts_with

### DISCUSSION THREAD

**Discussion by eposha** [prisma/prisma#17327] [Q&A]
Hi,

I have model (Postgres)
<img width="296" alt="image" src="https://user-images.githubusercontent.com/54452472/212570605-9ce804b7-a5dd-4c3e-9422-fbd5796f7e06.png">

In data base I save `email: [{ email: test@test.com }]`

How I can filter data by email with array_starts_with

I tried this (return empty array)

`  const prismaRequest: Prisma.ManagerFindManyArgs = {
        where: {
            authorId: id,
            email: {    path: ['email'],  array_starts_with: [searchByEmail] },
        },
  };`

and I tried this (return empty array)

`  const prismaRequest: Prisma.ManagerFindManyArgs = {
        where: {
            authorId: id,
            email: { path: ['email'], array_contains: [searchByEmail] },
        },
  };`

But working variant is only 

`  const prismaRequest: Prisma.ManagerFindManyArgs = {
        where: {
            authorId: id,
            email: { array_contains: [{ email: searchByEmail }] },
        },
  };`

But I need filter by array_starts_with

Thanks

**Comment by nurul3101** [comment]
Hello @eposha 👋 

Can you try defining the JSON as an object with a field having an array instead of storing directly an array in JSON field?

Here's what I tried:

Considering this schema:
```prisma
datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

generator client {
  provider        = "prisma-client-js"
}

model User {
  id      Int    @id @default(autoincrement())
  name    String
  details Json
  pets    Json
  Item    Item[]
}

model Organization {
  id    Int    @id @default(autoincrement())
  name  String
  items Item[]
}

model Item {
  id          Int           @id @default(autoincrement())
  ownedByUser Boolean
  owner       User?         @relation(fields: [ownerId], references: [id])
  ownerId     Int
  ownerOrg    Organization? @relation(fields: [ownerOrgId], references: [id])
  ownerOrgId  Int?
}
```
And this script:
```typescript
import { Prisma, PrismaClient } from "@prisma/client";

const prisma = new PrismaClient({
  log: ["query", "info", "warn"],
});

// A `main` function so that you can use async/await
async function main() {
  // ... you will write your Prisma Client queries here
  await prisma.user.deleteMany();

  //Create a new user with an item
  await prisma.user.create({
    data: {
      name: "John",
      id: 1,
      details: {
        name: ["John"],
      },
      pets: {
        sanctuaries: ["RSPCA"],
      },
    },
  });

  const getUsers = await prisma.user.findMany({
    where: {
      pets: {
        path: ["sanctuaries"],
        array_starts_with: "RSPCA",
      },
    },
  });

  console.log(getUsers);
}

main()
  .catch((e) => {
    throw e;
  })
  .finally(async () => {
    await prisma.$disconnect();
  });
```

Here's the response:
```
prisma:query SELECT "postgres"."User"."id", "postgres"."User"."name", "postgres"."User"."details", "postgres"."User"."pets" FROM "postgres"."User" WHERE ((("postgres"."User"."pets"#>ARRAY[$1]::text[])::jsonb->0)::jsonb = $2 AND JSONB_TYPEOF(("postgres"."User"."pets"#>ARRAY[$3]::text[])::jsonb) = $4) OFFSET $5
[
  {
    id: 1,
    name: 'John',
    details: { name: [Array] },
    pets: { sanctuaries: [Array] }
  }
]
```

Though I didn't pass an array in array_contains, I just passed value directly. It looks like maybe our documentation needs to be ipdated.

**Reply by nurul3101** [reply]
I have created a Docs Issue so that we can validate the correct behaviour: https://github.com/prisma/docs/issues/4413

**Comment by eposha** [comment]
@nurul3101 Thanks, I will try

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
